config/crd/bases/operator.antrea.vmware.com_antreainstalls.yaml

```diff
@@ -1,10 +1,10 @@
 
 ---
-apiVersion: apiextensions.k8s.io/v1
+apiVersion: apiextensions.k8s.io/v1beta1
```

Comment:
It seems that `apiextensions.k8s.io/v1beta1` API version of CustomResourceDefinition is no longer served as of v1.22. And `apiextensions.k8s.io/v1` is available since v1.16. As it's generated with controller-gen, I suggest to upgrade controller-gen and generate a new one.
[https://kubernetes.io/docs/reference/using-api/deprecation-guide/](https://kubernetes.io/docs/reference/using-api/deprecation-guide/)

Also, we may need to update this crd file under `deploy/kubernetes` and `deploy/openshift` as well if there is any update of the crd.

Reply:
@jwsui @salv-orlando
That CRD file exists in the following locations:
./config/crd/operator.antrea.vmware.com_antreainstalls.yaml
./config/crd/bases/operator.antrea.vmware.com_antreainstalls.yaml
./bundle/manifests/operator.antrea.vmware.com_antreainstalls.yaml
./packagemanifests/0.0.1/operator.antrea.vmware.com_antreainstalls.yaml

Plus the two under deploy/openshift and deploy/kubernetes.
I wonder if there is some way to tidy up this mess.

Any suggestions?

Reply:
Maybe recreate all of these with make manifests? Except for the one at:
 ./packagemanifests/0.0.1/operator.antrea.vmware.com_antreainstalls.yaml

as it's versioned?

Reply:
For `./config/crd/operator.antrea.vmware.com_antreainstalls.yaml` and `./config/crd/bases/operator.antrea.vmware.com_antreainstalls.yaml`, I agree with recreating them in one make target because `./config/crd/operator.antrea.vmware.com_antreainstalls.yaml` should be updated with command `kustomize build config/crd` when `./config/crd/bases/operator.antrea.vmware.com_antreainstalls.yaml` is updated. Also, the two under `deploy/openshift` and `deploy/kubernetes` should be updated as well.

For `./bundle/manifests/...` and `./packagemanifests`, I think it's another topic about `bundle format` and `package manifest format`. We already have make targets for them(`make bundle` and `make packagemanifests`), and maybe we can just keep it. Also, I'm not sure whether we need to maintain `./packagemanifests/...` as `package manifest format` is deprecated since OCP4.5. @salv-orlando any suggestions?